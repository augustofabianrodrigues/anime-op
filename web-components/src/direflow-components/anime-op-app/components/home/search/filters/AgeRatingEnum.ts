enum AgeRating {
  GeneralAudiences = 'G',
  ParentalGuidanceSuggested = 'PG',
  Restricted = 'R',

  // Explicit = 'R18', -- No, it won't be available ðŸ™ˆ
}

export default AgeRating;
