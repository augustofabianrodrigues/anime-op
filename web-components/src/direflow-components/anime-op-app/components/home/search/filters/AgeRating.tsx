import { Styled } from 'direflow-component';
import React, { FC, useState } from 'react';
import LabeledCheckbox from '../../../shared/LabeledCheckbox';
import styles from './AgeRating.less';
import AgeRatingEnum from './AgeRatingEnum';

const ageRatingLabels = {
  [AgeRatingEnum.GeneralAudiences.toString()]: 'General Audiences',
  [AgeRatingEnum.ParentalGuidanceSuggested.toString()]: 'Parental Guidance Suggest',
  [AgeRatingEnum.Restricted.toString()]: 'Restricted',
};

const AgeRating: FC = () => {
  // TODO: Uplift state
  const [values, setValues] = useState([
    AgeRatingEnum.GeneralAudiences,
    AgeRatingEnum.ParentalGuidanceSuggested,
  ]);

  const handleCheckboxChange = (type: AgeRatingEnum, value: boolean) => {
    setValues((prevValues) =>
      value ? [...prevValues, type] : prevValues.filter((v) => v !== type)
    );
  };

  const renderCheckbox = (type: AgeRatingEnum) => {
    return (
      <LabeledCheckbox
        label={ageRatingLabels[type]}
        value={values.includes(type)}
        onChange={(value) => handleCheckboxChange(type, value)}
      />
    );
  };

  return (
    <Styled styles={styles}>
      <div className="age-rating">
        <h3>Age Rating</h3>

        {renderCheckbox(AgeRatingEnum.GeneralAudiences)}
        {renderCheckbox(AgeRatingEnum.ParentalGuidanceSuggested)}
        {renderCheckbox(AgeRatingEnum.Restricted)}
      </div>
    </Styled>
  );
};

export default AgeRating;
