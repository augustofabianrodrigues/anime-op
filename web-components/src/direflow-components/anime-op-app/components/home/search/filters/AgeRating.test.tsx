import React from 'react';
import renderer from 'react-test-renderer';
import AgeRatingEnum from '../../../../models/AgeRatingEnum';
import AgeRating from './AgeRating';

it('matches snapshot of unchecked General Audiences, Parental Guidance Suggest, and Restricted', () => {
  const renderTree = renderer
    .create(<AgeRating values={[]} onChange={() => {}} />)
    .toJSON();
  expect(renderTree).toMatchSnapshot();
});

it('matches snapshot of unchecked General Audiences, and Parental Guidance Suggest and checked Restricted', () => {
  const renderTree = renderer
    .create(
      <AgeRating values={[AgeRatingEnum.Restricted]} onChange={() => {}} />
    )
    .toJSON();
  expect(renderTree).toMatchSnapshot();
});
