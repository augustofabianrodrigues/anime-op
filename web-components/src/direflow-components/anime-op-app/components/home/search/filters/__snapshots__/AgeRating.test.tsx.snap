// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot of unchecked General Audiences, Parental Guidance Suggest, and Restricted 1`] = `
<div
  className="age-rating"
>
  <h3>
    Age Rating
  </h3>
  <div
    className="labeled-checkbox"
    onClick={[Function]}
  >
    <div
      className="labeled-checkbox-control"
    >
      <input
        checked={false}
        onChange={[Function]}
        type="checkbox"
      />
      <svg
        fill="none"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          className="labeled-checkbox-control--check"
          d="M9 11L12 14L22 4"
          stroke="#7F9CF5"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
        />
        <path
          className="labeled-checkbox-control--square"
          d="M19 3H5C3.89543 3 3 3.89543 3 5V19C3 20.1046 3.89543 21 5 21H19C20.1046 21 21 20.1046 21 19V5C21 3.89543 20.1046 3 19 3Z"
          stroke="#FBFDFE"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
        />
      </svg>
    </div>
    <label>
      General Audiences
    </label>
  </div>
  <div
    className="labeled-checkbox"
    onClick={[Function]}
  >
    <div
      className="labeled-checkbox-control"
    >
      <input
        checked={false}
        onChange={[Function]}
        type="checkbox"
      />
      <svg
        fill="none"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          className="labeled-checkbox-control--check"
          d="M9 11L12 14L22 4"
          stroke="#7F9CF5"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
        />
        <path
          className="labeled-checkbox-control--square"
          d="M19 3H5C3.89543 3 3 3.89543 3 5V19C3 20.1046 3.89543 21 5 21H19C20.1046 21 21 20.1046 21 19V5C21 3.89543 20.1046 3 19 3Z"
          stroke="#FBFDFE"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
        />
      </svg>
    </div>
    <label>
      Parental Guidance Suggest
    </label>
  </div>
  <div
    className="labeled-checkbox"
    onClick={[Function]}
  >
    <div
      className="labeled-checkbox-control"
    >
      <input
        checked={false}
        onChange={[Function]}
        type="checkbox"
      />
      <svg
        fill="none"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          className="labeled-checkbox-control--check"
          d="M9 11L12 14L22 4"
          stroke="#7F9CF5"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
        />
        <path
          className="labeled-checkbox-control--square"
          d="M19 3H5C3.89543 3 3 3.89543 3 5V19C3 20.1046 3.89543 21 5 21H19C20.1046 21 21 20.1046 21 19V5C21 3.89543 20.1046 3 19 3Z"
          stroke="#FBFDFE"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
        />
      </svg>
    </div>
    <label>
      Restricted
    </label>
  </div>
</div>
`;

exports[`matches snapshot of unchecked General Audiences, and Parental Guidance Suggest and checked Restricted 1`] = `
<div
  className="age-rating"
>
  <h3>
    Age Rating
  </h3>
  <div
    className="labeled-checkbox"
    onClick={[Function]}
  >
    <div
      className="labeled-checkbox-control"
    >
      <input
        checked={false}
        onChange={[Function]}
        type="checkbox"
      />
      <svg
        fill="none"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          className="labeled-checkbox-control--check"
          d="M9 11L12 14L22 4"
          stroke="#7F9CF5"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
        />
        <path
          className="labeled-checkbox-control--square"
          d="M19 3H5C3.89543 3 3 3.89543 3 5V19C3 20.1046 3.89543 21 5 21H19C20.1046 21 21 20.1046 21 19V5C21 3.89543 20.1046 3 19 3Z"
          stroke="#FBFDFE"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
        />
      </svg>
    </div>
    <label>
      General Audiences
    </label>
  </div>
  <div
    className="labeled-checkbox"
    onClick={[Function]}
  >
    <div
      className="labeled-checkbox-control"
    >
      <input
        checked={false}
        onChange={[Function]}
        type="checkbox"
      />
      <svg
        fill="none"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          className="labeled-checkbox-control--check"
          d="M9 11L12 14L22 4"
          stroke="#7F9CF5"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
        />
        <path
          className="labeled-checkbox-control--square"
          d="M19 3H5C3.89543 3 3 3.89543 3 5V19C3 20.1046 3.89543 21 5 21H19C20.1046 21 21 20.1046 21 19V5C21 3.89543 20.1046 3 19 3Z"
          stroke="#FBFDFE"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
        />
      </svg>
    </div>
    <label>
      Parental Guidance Suggest
    </label>
  </div>
  <div
    className="labeled-checkbox"
    onClick={[Function]}
  >
    <div
      className="labeled-checkbox-control active"
    >
      <input
        checked={true}
        onChange={[Function]}
        type="checkbox"
      />
      <svg
        fill="none"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          className="labeled-checkbox-control--check"
          d="M9 11L12 14L22 4"
          stroke="#7F9CF5"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
        />
        <path
          className="labeled-checkbox-control--square"
          d="M19 3H5C3.89543 3 3 3.89543 3 5V19C3 20.1046 3.89543 21 5 21H19C20.1046 21 21 20.1046 21 19V5C21 3.89543 20.1046 3 19 3Z"
          stroke="#FBFDFE"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
        />
      </svg>
    </div>
    <label>
      Restricted
    </label>
  </div>
</div>
`;
